.gitlab-ci.yml

```diff
@@ -0,0 +1,100 @@
+##############################################################
+# Copyright 2023 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+##############################################################
+
+stages: 
+  - test
+
+## YAML Anchors
+.build-core: &build-core
+    - git clone https://github.com/flux-framework/flux-core
+    - cd flux-core
+    # Some tests parse the most recent tag for testing.
+    - lstopo --of xml >$(hostname).xml
+    - export FLUX_HWLOC_XMLFILE=$(pwd)/$(hostname).xml
```

Comment:
Well, as its currently used yes, but could`flux-sched` or other repos relying on hwloc take advantage of being able to read the topology from a file? (with some support needed from `flux-sched` or the repo in question)

Reply:
Yeah, that's a good point. I was thinking that the `FLUX_HWLOC_XMLFILE` should only be used when it has been shown to be necessary, but I don't think it will hurt to have it enabled by default in the gitlab ci builds.

Reply:
As an aside -- over half sched tests were failing when I ran `make check` on tioga yesterday. I wondered if it was the timeouts again... (didn't look super closely)